### PR TITLE
[NetAppFiles]update to 2022-05-01 add tests

### PR DESCRIPTION
**Pull request by audunn**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: eng/nunit.runsettings [hunk 1]
```diff
@@ -19,7 +19,7 @@
         <!-- <Parameter name="DisableAutoRecording" value="true" /> -->
 
         <!-- Change the test mode -->
-        <!-- <Parameter name="TestMode" value="Record" /> -->
+         <!--<Parameter name="TestMode" value="Record" />-->
```
Comment (ArthurMa1978) [inline eng/nunit.runsettings:22]: dont change this

File: sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Extensions/NetAppExtensions.cs [hunk 2]
```diff
@@ -130,6 +130,32 @@ public static Response<NetAppCheckAvailabilityResult> CheckNetAppQuotaAvailabili
...
 
+        /// <summary>
+        /// Provides storage to network proximity and logical zone mapping information.
+        /// Request Path: /subscriptions/{subscriptionId}/providers/Microsoft.NetApp/locations/{location}/regionInfo
+        /// Operation Id: NetAppResource_QueryRegionInfo
+        /// </summary>
+        /// <param name="subscriptionResource"> The <see cref="SubscriptionResource" /> instance the method will execute against. </param>
+        /// <param name="location"> The location. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        public static async Task<Response<RegionInfo>> QueryRegionInfoNetAppResourceAsync(this SubscriptionResource subscriptionResource, AzureLocation location, CancellationToken cancellationToken = default)
```
Comment (ArthurMa1978) [inline sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Extensions/NetAppExtensions.cs:141]: The name is weird, would like to have better name by using `override-operation-name`

File: sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Models/AccountEncryption.cs [hunk 3]
```diff
@@ -8,21 +8,29 @@
 namespace Azure.ResourceManager.NetApp.Models
 {
     /// <summary> Encryption settings. </summary>
-    internal partial class AccountEncryption
+    public partial class AccountEncryption
```
Comment (ArthurMa1978) [inline sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Models/AccountEncryption.cs:11]: `AccountEncryption` is too generic which may cause name collision with other Azure SDK, consider to prepend the RP name to `NetAppAccountEncryption`

File: sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Models/KeySource.cs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.NetApp.Models
+{
+    /// <summary> The encryption keySource (provider). Possible values (case-insensitive):  Microsoft.NetApp, Microsoft.KeyVault. </summary>
+    public readonly partial struct KeySource : IEquatable<KeySource>
```
Comment (ArthurMa1978) [inline sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Models/KeySource.cs:14]: too generic

File: sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Models/KeyVaultProperties.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+
+namespace Azure.ResourceManager.NetApp.Models
+{
+    /// <summary> Properties of key vault. </summary>
+    public partial class KeyVaultProperties
```
Comment (ArthurMa1978) [inline sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Models/KeyVaultProperties.cs:13]: lots of service contains this name, consider to prepend the `NetApp` to `NetAppKeyVaultProperties`

File: sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md [hunk 6]
```diff
@@ -8,8 +8,8 @@ azure-arm: true
 csharp: true
 library-name: NetApp
 namespace: Azure.ResourceManager.NetApp
-require: https://github.com/Azure/azure-rest-api-specs/blob/aa8a23b8f92477d0fdce7af6ccffee1c604b3c56/specification/netapp/resource-manager/readme.md
-tag: package-netapp-2022-03-01
+require: https://github.com/Azure/azure-rest-api-specs/blob/e31e3938529269e0e6a81f60b2fdc6d2aec5b9df/specification/netapp/resource-manager/readme.md
+tag: package-netapp-2022-05-01
```
Comment (ArthurMa1978) [inline sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md:12]: please run `eng/script/Export-API.ps1 netapp` to update the API view file

File: sdk/netapp/Azure.ResourceManager.NetApp/CHANGELOG.md [hunk 7]
```diff
@@ -3,9 +3,18 @@
 ## 1.0.0-beta.2 (Unreleased)
 
 ### Features Added
+    - Added QueryRegionInfoNetAppResourceAsync
```
Comment (ArthurMa1978) [inline sdk/netapp/Azure.ResourceManager.NetApp/CHANGELOG.md:6]: Should describe the upgrading to support new service api version 2022-05-01

File: sdk/netapp/Azure.ResourceManager.NetApp/tests/ResourceTests/ANFBackupPolicyTests.cs [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+                {
+                    await Task.Delay(40000);
+                }
+                await _netAppAccount.DeleteAsync(WaitUntil.Completed);
+            }
+            _resourceGroup = null;
+        }
+
+        [Test]
+        [RecordedTest]
```
Comment (ArthurMa1978) [inline sdk/netapp/Azure.ResourceManager.NetApp/tests/ResourceTests/ANFBackupPolicyTests.cs:64]: I'm not sure which one, but if there has race condition issue, can try `NonParallelizable`

File: sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Extensions/NetAppExtensions.cs (hunk 2 above)
Comment (audunn) [reply sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Extensions/NetAppExtensions.cs:141]: renamed

File: sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Models/KeySource.cs (hunk 4 above)
Comment (audunn) [reply sdk/netapp/Azure.ResourceManager.NetApp/src/Generated/Models/KeySource.cs:14]: renamed

File: sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md (hunk 6 above)
Comment (audunn) [reply sdk/netapp/Azure.ResourceManager.NetApp/src/autorest.md:12]: ran, did not seem to change much

File: sdk/netapp/Azure.ResourceManager.NetApp/tests/ResourceTests/ANFBackupPolicyTests.cs (hunk 8 above)
Comment (audunn) [reply sdk/netapp/Azure.ResourceManager.NetApp/tests/ResourceTests/ANFBackupPolicyTests.cs:64]: updated, seems to work now

File: eng/nunit.runsettings (hunk 1 above)
Comment (audunn) [reply eng/nunit.runsettings]: reverted

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**